azerothcore/azerothcore-wotlk
# Update Blackrock Drake path waypoints and movement template.

## Changes Proposed:
-  Updates the `creature_template_movement` and `waypoint_data` of the two [Blackrock Drakes](https://www.wowhead.com/wotlk/npc=8964/blackrock-drake) inside The Molten Span to more blizzlike paths and animation.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10177
- Closes https://github.com/chromiecraft/chromiecraft/issues/2789

## SOURCE:

Waypoints are collected from TBCC sniffs, no source on `creature_template_movement` changes nor the delay on [L26](https://github.com/azerothcore/azerothcore-wotlk/pull/12556/files#diff-b858d47f7e7bb16a580f0caa698eb6c6779b1e7ef0180a5cb3cd3759951a8ee2R26), though they appear more blizzlike due to those changes as well.


## Tests Performed:

- Tested in-game paths for both drakes.


## How to Test the Changes:


1. Use the command `.tele themoltenspan`.
2. Observe the two drakes following the paths correctly.

## Known Issues and TODO List:


- The drakes have issues transitioning between waypoints where their animation sort of resets for a small amount of time, which I believe is due to a lack of [smoothed out waypoints](https://github.com/azerothcore/azerothcore-wotlk/issues/10096), but that's not within the scope of this PR.
- The delay on [L26](https://github.com/azerothcore/azerothcore-wotlk/pull/12556/files#diff-b858d47f7e7bb16a580f0caa698eb6c6779b1e7ef0180a5cb3cd3759951a8ee2R26) needs to be tested and verified. If someone more familiar knows if that value can be found in a sniff parse, I can try to hunt it down, but I'm not familiar with the structure of sniffs enough to find it anywhere, so I just guessed there.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
